app/frontend/containers/CallToolView/CallToolView.js

```diff
@@ -147,18 +148,14 @@ class CallToolView extends Component {
     return _.sample(candidates);
   }
 
-  selectedTargetIndex() {
-    return _.findIndex(this.props.targets, this.state.selectedTarget);
-  }
-
   submit(event: any) {
     event.preventDefault();
     if(!this.validateForm()) return;
     this.setState({ errors: {}, loading: true });
     ChampaignAPI.calls.create({
       pageId: this.props.pageId,
       memberPhoneNumber: this.state.form.memberPhoneCountryCode + this.state.form.memberPhoneNumber,
-      targetIndex: this.selectedTargetIndex()
+      targetId: this.state.selectedTarget.id
```

Comment:
Flow's warning us that `this.state.selectedTarget` could be `undefined`.  From a quick glance, it seems this could happen if the component is mounted and there's an issue with the relevant props that we use to select the target (e.g. country) because it's dynamically set.

You'll need to check that `selectedTarget` is present just before the api call. 


Note: `this.setState` returns a new state object, so any checks will be invalid after that:

```js
if (!this.state.selectTarget) return; // flow can guarantee `selectedTarget` is present in this `this.state` object
this.setState({ loading: true }); // this.state is now a different state object so flow can't guarantee our previous check is still valid for this new state obj.
ChampaignAPI.calls.create({
  ...
  targetId: this.state.selectedTarget.id, // so flow will warn here
});
```
This would satisfy the type checker:
```js
this.setState({ loading: true });
if (!this.state.selectTarget) return;
ChampaignAPI.calls.create({
  ...
  targetId: this.state.selectedTarget.id,
});
```

... but if we returned before making the api call, our loading state would just get stuck in `loading: true`.

So this is another option: 

```js
if (!this.state.selectTarget) return;
ChampaignAPI.calls.create({
  ...
  targetId: this.state.selectedTarget.id,
});
this.setState({ loading: true });
```